### PR TITLE
EIB: Jenkibs job not getting triggered after jenkins uplift to 2.222.3

### PR DESCRIPTION
**Pull request by Surya-Kolli**
### Description of the Change
With jenkins uplift to 2.222.3, we have CSRF protection which needs crumbs to make rest post calls to jenkins.
The crumb issuer end point is not working which will be hitted when BASIC_AUTH_CSRF_PROTECTION is the AUTH_TYPE
for the subscrptions to trigger the jenkins jobs. Due to this the jenkins obs are not getting triggered. 
Made the code change that triggers the jenkins jobs for the subscrptions for AUTH_TYPE "BASIC_AUTH_CSRF_PROTECTION"

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: K.Suryakumari suryakumari.kolli@ericsson.com

### CODE REVIEW

File: src/main/java/com/ericsson/ei/notifications/JenkinsCrumb.java [hunk 1]
```diff
@@ -95,7 +95,14 @@ public JsonNode fetchJenkinsCrumb(String encoding, String url)
      */
     private URL buildJenkinsCrumbUrl(String url) throws MalformedURLException {
         String baseUrl = urlParser.extractBaseUrl(url);
-        URL jenkinsCrumbUrl = new URL(baseUrl + JENKINS_CRUMB_ENDPOINT);
+        String contextPath = urlParser.extractContextPath(url).split("/")[1];
```
Comment (SantoshNC68) [inline src/main/java/com/ericsson/ei/notifications/JenkinsCrumb.java:98]: Final

File: src/main/java/com/ericsson/ei/notifications/JenkinsCrumb.java [hunk 2]
```diff
@@ -95,7 +95,14 @@ public JsonNode fetchJenkinsCrumb(String encoding, String url)
...
+        URL jenkinsCrumbUrl;
+        if (contextPath.equals("job")) {
+            jenkinsCrumbUrl = new URL(baseUrl + JENKINS_CRUMB_ENDPOINT);
+        } else {
+            String jenkinsBaseUrl = String.format("%s/%s", baseUrl, contextPath);
```
Comment (SantoshNC68) [inline src/main/java/com/ericsson/ei/notifications/JenkinsCrumb.java:103]: Final

File: src/main/java/com/ericsson/ei/notifications/JenkinsCrumb.java [hunk 3]
```diff
@@ -95,7 +95,14 @@ public JsonNode fetchJenkinsCrumb(String encoding, String url)
      */
     private URL buildJenkinsCrumbUrl(String url) throws MalformedURLException {
         String baseUrl = urlParser.extractBaseUrl(url);
-        URL jenkinsCrumbUrl = new URL(baseUrl + JENKINS_CRUMB_ENDPOINT);
+        String contextPath = urlParser.extractContextPath(url).split("/")[1];
+        URL jenkinsCrumbUrl;
+        if (contextPath.equals("job")) {
```
Comment (SantoshNC68) [inline src/main/java/com/ericsson/ei/notifications/JenkinsCrumb.java:100]: "job".equals(contextpath)

**Review by SantoshNC68 — APPROVED** [review]

File: pom.xml [hunk 4]
```diff
@@ -6,7 +6,7 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>com.github.ericsson</groupId>
     <artifactId>eiffel-intelligence</artifactId>
-    <version>3.0.1-SNAPSHOT</version>
+    <version>3.0.2-SNAPSHOT</version>
```
Comment (tobiasake) [inline pom.xml:9]: Have we released 3.0.1? What version number do we plan to use in next release?

Comment (Divya-Kuppula) [reply pom.xml:9]: 3.0.1 was not releases, my mistake will revert the pom.xml

**Review by tobiasake — APPROVED** [review]

**Pull request merged**